plugins/inputs/http_response/http_response.go

```diff
@@ -130,15 +130,21 @@ func (h *HTTPResponse) httpGather() (map[string]interface{}, error) {
 	// Start Timer
 	start := time.Now()
 	resp, err := h.client.Do(request)
+	fields["response_time"] = time.Since(start).Seconds()
+
 	if err != nil {
+		fields["http_response_code"] = -1
+		fields["response_string_match"] = 0
 		if h.FollowRedirects {
-			return nil, err
+			log.Printf("E! Failed to open HTTP connection: %s", err)
+			return fields, nil
```

Comment:
The other option would be to give the the fields and the errors one level up and logging them. I thought about that and my reasoning is the following:
Currently we either return the fields or the error and this would break this pattern. In my opinion this is a little easier to read. However I don't have a strong opinion on it. I can change it, if it's important for you.